### PR TITLE
feat: Add action for fetching identity token

### PR DESCRIPTION
**Pull request by akovtun-extenda**

### CODE REVIEW

File: identity-token/action.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    required: true
+  service-account:
+    description: Service account to print the identity token for. Full service account name needs to be specified.
+    required: true
+  service-account-key:
+    description:
+      The service account key which will be used for authentication.
+      The account key should be a base64 encoded JSON key stored as a secret.
+    required: true
+runs:
```
Comment (sasjo) [inline identity-token/action.yml:16]: The action should also specify an `outputs` section containing the `identity-token` according to the README.

File: identity-token/src/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const action = async () => {
+  const serviceAccountKey = core.getInput('service-account-key', { required: true });
+  const serviceAccount = core.getInput('service-account', { required: true });
+  const audiences = core.getInput('audiences', { required: true });
+
+  await setupGcloud(serviceAccountKey, process.env.GCLOUD_INSTALLED_VERSION || 'latest');
+  const token = await fetchToken(serviceAccount, audiences);
+  core.setSecret(token);
+  core.exportVariable('IDENTITY_TOKEN', token);
+  core.setOutput('iam-token', token);
```
Comment (sasjo) [inline identity-token/src/index.js:15]: The output should be `identity-token` according to the README.

**Review by sasjo — CHANGES_REQUESTED** [review]
The output variable should be fixed. Is the safe-guards to only allow this in staging enforced inside GCP using IAM permissions or should the action also check something on the SA?

**Comment by akovtun-extenda** [comment]
None of the ci-cd-pipeline SAs have a token creator role which is required to be able to generate tokens. This role will be granted only in staging.

**Review by Santos92 — APPROVED** [review]

**Comment by akovtun-extenda** [comment]
do not merge, please. Need to update the project's module first, to add a needed role for the ci-cd SA.

**Review by sasjo — APPROVED** [review]
LGTM

**Pull request merged**